src/libraries/System.Runtime.Serialization.Formatters/tests/TestConfiguration.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Runtime.Serialization.Formatters.Tests
+{
+    public static class TestConfiguration
+    {
+        internal const string EnableBinaryFormatterSwitchName = "System.Runtime.Serialization.EnableUnsafeBinaryFormatterSerialization";
+
+        public static readonly bool IsBinaryFormatterSupported;
+        public static readonly bool IsBinaryFormatterEnabled;
+        public static readonly bool IsFeatureSwitchRespected;
+
+        static TestConfiguration()
+        {
+            if (!PlatformDetection.IsBinaryFormatterSupported)
+            {
+                return;
+            }
+
+            IsBinaryFormatterSupported = true;
+            IsBinaryFormatterEnabled = true;
+
+            if (PlatformDetection.IsNetFramework)
+            {
+                Console.WriteLine("BinaryFormatter is always enabled by the platform (netfx).");
+                return;
+            }
+
+            IsFeatureSwitchRespected = true;
+
+            if (!AppContext.TryGetSwitch(EnableBinaryFormatterSwitchName, out IsBinaryFormatterEnabled))
+            {
+                IsBinaryFormatterEnabled = false;
+            }
```

Comment:
I'm not completely following this logic. We'll still be running the BinaryFormatter tests as part of CI, yes?

Reply:
If the tests get the 8.1 version of the assembly: no.  If yes, then it is identifying if the tests have pre-set the AppContext switch, so whether it should expect success or failure

Reply:
Who sets the switch? Is it automatically set in some test suite?

High-level, will the tests be running in CI automatically for the netcoreappcurrent version of the assembly, as they do today?

Reply:
> Who sets the switch?

We set it by default for our test projects via a props file, but this test suite overrides it to turn it off... except that netfx doesn't have the switch, so it's not a simple "assume it isn't going to work".

> High-level, will the tests be running in CI automatically for the netcoreappcurrent version of the assembly, as they do today?

Yes.  They'll _also_ run this library under netcoreappminimum if #54639 ever happens.  If that stays as "well, we haven't done over the past 5 years, why would we right now?" then we'll come up with some other plan (such as giving this project a PackageReference to BinaryFormatter, once there's a package to reference)